controllers/onload_controller.go

```diff
@@ -206,102 +191,77 @@ func createModule(
 	return module, nil
 }
 
-func (r *OnloadReconciler) createControlPlaneDaemonSet(
+func (r *OnloadReconciler) createDevicePluginDaemonSet(
```

Comment:
I'm not sure if this would be helpful or not, but have you considered breaking down this function into a few smaller ones?
Currently it takes up 226 out of 420 lines in this file.

My thought is that we could add a couple functions like:
```
func createDevicePluginPod(onload *Onload) corev1.Pod
func createWorkerPod(onload *Onload) corev1.Pod
```
etc.

Reply:
Yeah, `createDevicePluginDaemonSet()` is quite a long function, not due to its complexity but rather because of how it's written, as @pcolledg-amd says. If we split it, we would need to carry the volume mounts (and maybe something else) across as a set of parameters to these subroutines, which would be another inconvenience.

@tcrawley-xilinx Does it sound convincing?

Reply:
I think leave it as it is